### ISSUE TITLE
SMTP test works, but sending test email in campaign fails

### ISSUE THREAD

**Issue by AkashRajpurohit** [knadh/listmonk#1145]
**Version:**
 - listmonk: v2.3.0
 - OS: Debian 11

**Description of the bug and steps to reproduce:**
When setting up the SMTP configuration, the test email goes fine and is received to the recipient email address. However, when I try to send the test email from the campaign "send test message" section, the email sending fails and error is visible on the logs section.

**Screenshots:**
![SMTP Settings test email works](https://user-images.githubusercontent.com/30044630/212333284-5210374e-bbcb-414e-9a62-8f0d4b04da5d.png)

![Test message from campaign](https://user-images.githubusercontent.com/30044630/212333797-81ba946b-ff43-4e23-a7af-77af71113409.png)

![Error when message send from campaign](https://user-images.githubusercontent.com/30044630/212333868-56384030-0d9d-4839-a158-d226843e8e21.png)

**Comment by knadh** [comment]
`EOF` happens when there's something wrong with the SMTP server settings (typically wrong TLS settings). The message is indeed ambiguous. Will look into that.

**Comment by AkashRajpurohit** [comment]
Hi @knadh I was able to figure out about the EOF error when I had wrong config for ZOHO STMP earlier on the "Settings -> SMTP" page (first image), but once I fixed that, I was able to successfully send a test email from that section.

However, with the same config, it is failing to send test email from the campaign send test email section.

I have verified that the SMTP config is saved correctly on the postgres db. So I don't think this could be an error with the SMTP config. Thoughts?
